internal/pkg/describe/stack/stack.go

```diff
@@ -0,0 +1,227 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stack
+
+import (
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/ecs"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/aws/session"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/deploy/cloudformation/stack"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/store"
+	"github.com/aws/aws-sdk-go/aws"
+	clientsession "github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+)
+
+const (
+	// Ignored resources
+	rulePriorityFunction = "Custom::RulePriorityFunction"
+	waitCondition        = "AWS::CloudFormation::WaitCondition"
+	waitConditionHandle  = "AWS::CloudFormation::WaitConditionHandle"
+
+	serviceLogicalID = "Service"
+)
+
+type ecsService interface {
+	TaskDefinition(taskDefName string) (*ecs.TaskDefinition, error)
+}
+
+type storeSvc interface {
+	archer.EnvironmentGetter
+	archer.EnvironmentLister
+}
+
+type stackDescriber interface {
+	DescribeStacks(input *cloudformation.DescribeStacksInput) (*cloudformation.DescribeStacksOutput, error)
+	DescribeStackResources(input *cloudformation.DescribeStackResourcesInput) (*cloudformation.DescribeStackResourcesOutput, error)
+}
+
+type sessionFromRoleProvider interface {
+	FromRole(roleARN string, region string) (*clientsession.Session, error)
+}
+
+// Describer retrieves information of a CloudFormation Stack.
+type Describer struct {
+	project string
+
+	store           storeSvc
+	ecsClient       map[string]ecsService
+	stackDescribers map[string]stackDescriber
+	sessProvider    sessionFromRoleProvider
+}
+
+// NewDescriber instantiates a new Describer.
+func NewDescriber(projectName string) (*Describer, error) {
+	svc, err := store.New()
+	if err != nil {
+		return nil, fmt.Errorf("connect to store: %w", err)
+	}
+	return &Describer{
+		project:         projectName,
+		store:           svc,
+		stackDescribers: make(map[string]stackDescriber),
+		ecsClient:       make(map[string]ecsService),
+		sessProvider:    session.NewProvider(),
+	}, nil
+}
+
+// EnvVars returns the environment variables of the task definition.
+func (d *Describer) EnvVars(env *archer.Environment, appName string) ([]*EnvVars, error) {
```

Comment:
mmm, although the package name is `stack` we're using an ECS client to retrieve this information. Why do we need to create a new package?

Can we not have these methods part of a private struct inside the `describe` pkg? Maybe something like an `appDescriber` which takes as input to the constructor a project, env, and app name and then all of its methods don't need to take any parameters.

What do you think?

Reply:
Creating this package is because previously `WebAppDescriber` is very big and when I was trying to add `Describe()` method to it, I found it was hard to test the function because it uses too many functions within the same struct. Then I found most of the dependency functions for `Describe()` are actually not related to "describing web app", they are more about getting stack info, which is more generic and can be applied to not only application but also environment (for env show). That's the reason why I created a new package.

However, I do agree that maybe `EnvVars()` should not belong to this package or we can rename this package name. What do you think?

Reply:
After offline discussion, I'll remove stack package and add another struct `AppDescriber` depending on `stackDescriber` to perform general app related function as an info provider for `WebAppDescriber`.